docs/gettinginvolved/plugins/building-driver.rst

```diff
@@ -0,0 +1,161 @@
+
+Examples of status images include `Travis CI <http://docs.travis-ci.com/user/status-images/>`_ and `Jenkins <https://wiki.jenkins-ci.org/display/JENKINS/Embeddable+Build+Status+Plugin>`_.
+
+You should also clearly indicate which version of Flocker your driver has been certified against.
+
+
+Enabling Flocker users to install your storage driver
+=====================================================
+
+Once you've implemented your storage backend you'll want to allow Flocker users to use your package.
+The basic implementation strategy is that your user installs a Python package with your backend implementation on all Flocker nodes:
+
+.. prompt:: bash $
+
+    /opt/flocker/bin/pip install https://example.com/your/storageplugin-1.0.tar.gz
+
+You can also provide RPMs or DEBs that have same effect of installing a new Python package.
+
+Once your users have installed the package, they will write an :file:`agent.yml` file (:file:`/etc/flocker/agent.yml`), whose ``backend`` key in the ``dataset`` section is the importable name of the Python package you've installed.
+
+All other sub-keys of the ``dataset`` section will be passed to a function you must implement (see below), and can be used to configure the resulting ``IBlockDeviceAPI`` instance.
+
+Typical parameters are authentication information or server addresses; whatever is necessary to configure your class.
+
+For example, if you installed a Python package importable ``mystorage_flocker_plugin``, and you require a username and password in order to log in to your storage system, you might tell your users to write a :file:`agent.yml` that looks like this:
+
+.. code-block:: yaml
+
+   version: 1
+     control-service:
+       hostname: "user.controlserver.example.com"
+     dataset:
+       backend: "mystorage_flocker_plugin"
+       username: "username_for_mystorage"
+       password: "abc123"
+
+Your :file:`mystorage_flocker_plugin/__init__.py` module needs to have a ``FLOCKER_BACKEND`` attribute with a ``flocker.node.BackendDescription`` instance, which will include a reference to factory function that constructs a ``IBlockDeviceAPI`` instance.
+
+The factory function will be called with whatever parameters the ``dataset`` section in :file:`agent.yml` is configured with.
+In the above example, that would be ``username`` and ``password``.
+
+Here's what the module could look like:
+
+.. code-block:: python
+
+    from flocker.node import BackendDescription, DeployerType
+    from mystorage_flocker_plugin._backend import MyStorageAPI
+
+    def api_factory(cluster_id, **kwargs):
+        return MyStorageAPI(cluster_id=cluster_id, username=kwargs[u"username"],
+                            password=kwargs[u"password"])
+
+    FLOCKER_BACKEND = BackendDescription(
+        name=u"mystorage_flocker_plugin", # name isn't actually used for 3rd party plugins
```

Comment:
This comment is confusing as the name has been used in the `agent.yml` example above.